mlir/lib/Ion/Transforms/CMakeLists.txt

```
@@ -34,6 +34,13 @@ FetchContent_Declare(
 )
 FetchContent_MakeAvailable(tomlplusplus)
 
+# Fetch json utilities
+FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz)
```

```suggestion
FetchContent_Declare(
  json
  GIT_REPOSITORY https://github.com/nlohmann/json.git
  GIT_TAG        v3.11.3
)
```
Can we use the git-tag checkout instead of pulling in the URL? If you want to use the URL, a hash should be provided to validate it.

---

I was going with git tag originally, but the package doc says it favors URL: https://github.com/nlohmann/json?tab=readme-ov-file#embedded-fetchcontent

---

https://github.com/PennyLaneAI/catalyst/pull/1516/commits/ea99d30b44359d8ab762496af1c2fe1b7f4f78cb

---

I noticed fetching in git source is significantly slower than downloading the tar ball. I will try to add a hash.

---

Using hash here https://github.com/PennyLaneAI/catalyst/pull/1516/commits/349c6a4531ef9676a2942e673868782d5d6e2c5f